wsrep_api.h

```diff
@@ -412,21 +419,23 @@ typedef enum wsrep_cb_status (*wsrep_sst_request_cb_t) (
  * Must support brute force applying for multi-master operation
  *
  * @param recv_ctx receiver context pointer provided by the application
- * @param data     data buffer containing the writeset
- * @param size     data buffer size
  * @param flags    WSREP_FLAG_... flags
  * @param meta     transaction meta data of the writeset to be applied
+ * @param data     data buffer containing the writeset
+ * @param err_buf  buffer containing error info (null/empty for n error)
+ * @param err_len  error buffer size
  *
  * @return error code:
  * @retval 0 - success
  * @retval non-0 - application-specific error code
  */
 typedef int (*wsrep_apply_cb_t) (
-    void*                     recv_ctx,
-    const void*               data,
-    size_t                    size,
-    uint32_t                  flags,
-    const wsrep_trx_meta_t*   meta
+    void*                   recv_ctx,
+    uint32_t                flags,
+    const wsrep_buf_t*      data,
+    const wsrep_trx_meta_t* meta,
+    void**                  err_buf,
```

Comment:
Is the application responsible for allocating ```err_buf```? If yes, whose responsibility it is to free it?